numpy/core/src/npysort/quicksort.c.src

```diff
@@ -69,10 +90,12 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
     @type@ *stack[PYA_QS_STACK];
     @type@ **sptr = stack;
     @type@ *pm, *pi, *pj, *pk;
-    npy_intp depth_limit = npy_get_msb(num) * 2;
+    int depth[PYA_QS_STACK];
+    int * psdepth = depth;
+    int cdepth = NPY_SORT_MIN(PYA_QS_STACK, npy_get_msb(num) * 2);
```

Comment:
Yep, cdepth can be anything that serves as a reasonable cutoff. It is independent of the actual stack depth as there may be jumps in it's value greater than one along the stack: partition gets popped, gets replaced by partition with cdepth decremented. Continuous bad partitioning actually leads to the smallest stack.